service/matching/matchingEngine.go

```diff
@@ -296,6 +296,21 @@ func (e *matchingEngineImpl) AddDecisionTask(
 		return false, err
 	}
 
+	// get the domainName
+	domainName, err := e.domainCache.GetDomainName(domainID)
+	if err != nil {
+		return false, err
+	}
+
+	// tell if the tasklist is partitioned or sticky
+	if int32(request.GetTaskList().GetKind()) == 0 && request.ForwardedFrom == "" {
```

Comment:
taskListKind was retrieved at the beginning of the method, you can just use it

Reply:
Resolved in newest commit